src/main/java/de/chojo/repbot/listener/voting/ReputationVoteListener.java

```diff
@@ -0,0 +1,118 @@
+    private final ReputationService reputationService;
+    private final ILocalizer loc;
+    private final Map<Long, VoteRequest> voteRequests = new HashMap<>();
+
+    public ReputationVoteListener(ReputationService reputationService, ILocalizer localizer) {
+        this.reputationService = reputationService;
+        this.loc = localizer;
+    }
+
+    @Override
+    public void onButtonClick(@NotNull ButtonClickEvent event) {
+        event.deferEdit().queue();
+        if (event.getButton() == null || event.getButton().getId() == null) return;
+
+        var matcher = VOTE.matcher(event.getButton().getId());
+        if (!matcher.find()) return;
+        if (!voteRequests.containsKey(event.getMessageIdLong())) return;
+        var voteRequest = voteRequests.get(event.getMessageIdLong());
+        if (!Verifier.equalSnowflake(voteRequest.member(), event.getMember())) return;
+        if ("vote:delete".equals(event.getButton().getId())) {
+            event.getMessage().delete().queue();
+            voteRequests.remove(event.getMessageIdLong());
+            return;
+        }
+
+        var target = voteRequest.getTarget(event.getButton().getId());
+
+        if (reputationService.submitReputation(event.getGuild(), event.getUser(), target.get().getUser(), voteRequest.refMessage(), null, ThankType.REACTION)) {
+            voteRequest.voted();
+            voteRequest.remove(event.getButton().getId());
+            voteRequest.voteMessage().
+                    editMessage(voteRequest.getNewEmbed(loc.localize("listener.messages.request.descrThank"
+                            , event.getGuild(), Replacement.create("MORE", voteRequest.remainingVotes()))))
+                    .setActionRows(getComponentRows(voteRequest.components()))
+                    .queue();
+            if (voteRequest.remainingVotes() == 0) {
+                voteRequest.voteMessage().delete().queueAfter(5, TimeUnit.SECONDS,
+                        suc -> voteRequests.remove(voteRequest.voteMessage().getIdLong()), err -> {
+                        });
+            }
+        }
+    }
+
+    public void registerVote(Message message, List<Member> members) {
+        var builder = new LocalizedEmbedBuilder(loc, message.getGuild())
+                .setTitle("listener.messages.request.title")
+                .setDescription("listener.messages.request.descr")
+                .setColor(Color.orange)
+                .setFooter(loc.localize("messages.destruction", message.getGuild(), Replacement.create("MIN", 1)));
+
+        Map<String, VoteComponent> components = new LinkedHashMap<>();
+
+        for (var member : members) {
+            var id = "vote:" + member.getIdLong();
+            components.put(id, new VoteComponent(member, Button.of(ButtonStyle.PRIMARY, id, member.getEffectiveName())));
+        }
+
+        var collect = components.values().stream().map(VoteComponent::component).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Why not `Collectors.toList()` or `Collectors.toUnmodifiableList()`

Reply:
Oh yeah... needed an array list in an earlier implementation, because I needed to add the deletion button.